src/Impostor.Server/Events/Game/BeforeGameCreatedEvent.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Impostor.Api.Events;
+using Impostor.Api.Games;
+using Impostor.Api.Innersloth;
+using Impostor.Server.Net.Redirector;
+using Impostor.Server.Net.State;
+
+namespace Impostor.Server.Events
+{
+    public class BeforeGameCreatedEvent : IBeforeGameCreatedEvent
+    {
+        private readonly ICollection<int> _gameKeys;
+
+        public BeforeGameCreatedEvent(ICollection<int> gameKeys)
+        {
+            _gameKeys = gameKeys;
+        }
+
+        public GameCode? GameCode { get; private set; }
+
+        public bool TryToSetCode(GameCode gameCode)
+        {
+            if (gameCode.Code.Length != 6)
+            {
+                throw new ArgumentException("GameCode must be 6 characters longs.");
+            }
+
+            if (!Regex.IsMatch(gameCode.Code, @"^[A-Z]+$"))
```

Comment:
Again, this would never return false, it would just crash in https://github.com/Impostor/Impostor/blob/dc759b1d573b59ff5c39dd2792c40ebdec7649a2/src/Impostor.Api/Innersloth/GameCodeParser.cs#L98-L111